scenes/ink_combat/inkwell/inkwell.gd

```diff
@@ -0,0 +1,49 @@
+class_name Inkwell
+extends StaticBody2D
+
+const SPLASH: PackedScene = preload("res://scenes/ink_combat/splash/splash.tscn")
+
+@export_enum("Cyan", "Magenta", "Yellow", "Black") var ink_color_name: int = 0
+
+var ink_amount: float = 0.
+
+@onready var animated_sprite_2d: AnimatedSprite2D = %AnimatedSprite2D
+@onready var animation_player: AnimationPlayer = %AnimationPlayer
+@onready var interact_label: Control = %InteractLabel
+
+
+func _ready() -> void:
+	var color: Color = InkBlob.INK_COLORS[ink_color_name]
+	animated_sprite_2d.modulate = color
+
+	var current_scene: Node = Engine.get_main_loop().current_scene
+	var screen_overlay: CanvasLayer = current_scene.get_node("ScreenOverlay")
+	interact_label.reparent(screen_overlay)
+	interact_label.label_text = InkBlob.InkColorNames.keys()[ink_color_name]
+
+
+func got_hit(
+	ink_blob_global_position: Vector2, _hit_vector: Vector2, ink_blob_color_name: int
+) -> void:
+	var splash: Node2D = SPLASH.instantiate()
+	splash.ink_color_name = ink_blob_color_name
+	Engine.get_main_loop().current_scene.add_child(splash)
+	splash.global_position = ink_blob_global_position
+
+
+func _get_ink_drinkers() -> Array[Node]:  #Array[InkDrinker]:
```

Comment:
I wish Godot could type this as `Array[InkDrinker]` 🙏🏾 . 

I'm not sure if it'd work, but maybe you can cast it when returning it?
Otherwise, I would remove the comment and add godot docs comments to explain this is expected to return `InkDrinker`s
https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_documentation_comments.html

Reply:
Ouch I didn't meant to leave this comment. Initially I had `Array[InkDrinker]` but Godot complained with a runtime error: `Trying to return an array of type "Array[Node]" where expected return type is "Array[InkDrinker]".`

I now tried casting `return enemies.filter(is_ink_drinker) as Array[InkDrinker]` but the same error happens:
![Captura desde 2025-03-31 14-46-51](https://github.com/user-attachments/assets/900458f2-6c6a-429a-b0b5-fe1e15750b1a)

So I removed the comment for now.